### DISCUSSION TITLE
Testing the model for the first time. The simplest event

### DISCUSSION THREAD

**Discussion by jarufe** [oreplay/oreplay#8] [Show and tell]
Sequence of data transfers in a Foot-O single-stage event with radiocontrols managed with OE
The following is not a complete pseudocode but the main steps

- In the previous days, the administrator creates the event and the stage
  - He doesn't provide any other previous configuration of classes, courses, controls, etc.

- Before the start, the administrator sends XML of entries with their start times
  - He creates XML from start list by classes
  - The client sends the XML
  - The server processes the XML and creates the following data:
    - classes, courses
    - IOF XML supports adding radiocontrols info, through Class tag -> RaceClass (0..*) -> OnlineControl (0..*). OE doesn't includes this information in the XML
    - runners, clubs
      - For each runner, create one record in runner_results and write the start time there. The result type is 0=Overall as the event is one-stage

- During competition, the administrator sends successive XML with intermediate times
  - He creates XML from Speaker, intermediate results
    - Initially, before any intermediate data arrives, a list can be obtained in which all the categories appear, all the runners, their start times, but without times. The XML will contain all the information. In the Result tag there will be a start time tag and another Status tag with the value "Inactive" (to indicate that he/she has not reached the finish line). For each split there will be an attribute status="Missing" and no time
    - In subsequent steps, as data arrive through the radio controls, the "gaps" in that initial report are filled. Where there is data, there will be no status="Missing" attribute and there will be time
    - When someone reaches the finish line, in the Result tag there will be the start time, finish time, position and status tags (here, since there is no course checking, it will be Ok)
  - The client sends the XML
  - The server processes the XML and creates/updates the following data:
    - If there were no classes, courses, controls, classes_controls, runners, clubs, they are created. If needed, it creates a runner_results record as in the first step
    - If it comes: finish_time, time, time_behind, position and status, it writes them in runner_results
    - If split data are coming, searches - updates/creates the record for each split
      - The sequence given by order_number field can be composed following the order of the controls in the XML

- During the competition, data from downloads may be sent periodically
  - The administrator creates the XML from Results, breakdown
  - The client sends the XML
  - The server processes the XML and creates/updates the following data:
    - If there were no classes, courses, controls, classes_controls, runners, clubs, they are created. If needed, it creates a runner_results record as in the first step
    - If it comes: finish_time, time, time_behind, position and status, it writes them in runner_results
    - If split data are coming, this time it will be the whole sequence of controls in the course, not only radiocontrols. Searches - updates/creates the record for each split
      - Reorder the sequence given by order_number field, as it will be new splits and other existing (radiocontrols)
      - If an existing one already had time but the new one says "Missing", don't update if it's a runner that has not finished yet
      
**ATTENTION**
- If a course has the same radiocontrol that is revisited, how to distinguish?
  - Checked. OE does not compare times to know if it could be the first visit or the second. For example, course 31, 32, 33, 100, 32. Radiocontrols 32 and 100. A runner does not pass through 32 the first time (he does not visit the control or sending the data does not work), he passes through 100 and then through 32. Although the passage time of 32 is greater than that of 100, OE does not interpret that it is the second visit but the first
- If the event has been configured so that finish time can be taken from radio control, the people's data will appear in the provisional results with partials if they have already reached the finish line but have not yet downloaded, and the splits will be status="Missing"

Sequence of XML files obtained during the test (there were no final download of SiCards so it's incomplete)
[OE12_SimplestTest.zip](https://github.com/oreplay/oreplay/files/13950415/OE12_SimplestTest.zip)

**Comment by freefri** [comment]
Example json. [00_StartsTest.json](https://github.com/oreplay/oreplay/files/14407332/00_StartsTest.json)

This will replace the step: 

"_Before the start, the administrator sends XML of entries with their start times_"

**Reply by freefri** [reply]
relates to https://github.com/oreplay/desktop-client/issues/1
